src/main/java/gregtech/common/metatileentities/multi/multiblockpart/MetaTileEntityFuelRodImportHatch.java

```diff
@@ -0,0 +1,251 @@
+                .label(10, 5, getMetaFullName());
+
+        builder.widget(new SlotWidget(importItems, 0, 79, 18, true, true)
+                .setBackgroundTexture(GuiTextures.SLOT));
+
+        return builder.bindPlayerInventory(player.inventory, GuiTextures.SLOT, 7, 60);
+    }
+
+    @Override
+    public void update() {
+        super.update();
+        if (!getWorld().isRemote && getOffsetTimer() % 5 == 0) {
+            pullItemsFromNearbyHandlers(getFrontFacing());
+        }
+    }
+
+    @Override
+    protected ModularUI createUI(EntityPlayer entityPlayer) {
+        return createUITemplate(entityPlayer).build(getHolder(), entityPlayer);
+    }
+
+    @Override
+    public void renderMetaTileEntity(CCRenderState renderState, Matrix4 translation, IVertexOperation[] pipeline) {
+        super.renderMetaTileEntity(renderState, translation, pipeline);
+        if (shouldRenderOverlay()) {
+            SimpleOverlayRenderer renderer = isExportHatch ? Textures.PIPE_OUT_OVERLAY : Textures.PIPE_IN_OVERLAY;
+            renderer.renderSided(getFrontFacing(), renderState, translation, pipeline);
+            SimpleOverlayRenderer overlay = isExportHatch ? Textures.ITEM_HATCH_OUTPUT_OVERLAY :
+                    Textures.ITEM_HATCH_INPUT_OVERLAY;
+            overlay.renderSided(getFrontFacing(), renderState, translation, pipeline);
+        }
+    }
+
+    @Override
+    public MultiblockAbility<IFuelRodHandler> getAbility() {
+        return MultiblockAbility.IMPORT_FUEL_ROD;
+    }
+
+    @Override
+    public void registerAbilities(List<IFuelRodHandler> abilityList) {
+        abilityList.add(this);
+    }
+
+    @Override
+    public boolean checkValidity(int depth) {
+        this.pairedHatch = getExportHatch(depth);
+        return pairedHatch != null;
+    }
+
+    @Override
+    public void readFromNBT(NBTTagCompound data) {
+        super.readFromNBT(data);
+        getLockedImport().setLock(data.getBoolean("locked"));
+        if (data.hasKey("partialFuel"))
+            this.partialFuel = GregTechAPI.materialManager.getMaterial(data.getString("partialFuel"));
+    }
+
+    @Override
+    public NBTTagCompound writeToNBT(NBTTagCompound data) {
+        data.setBoolean("locked", getLockedImport().isLocked());
+        if (partialFuel != null)
+            data.setString("partialFuel", this.partialFuel.toString());
+        return super.writeToNBT(data);
+    }
+
+    @Override
+    public void writeInitialSyncData(PacketBuffer buf) {
+        super.writeInitialSyncData(buf);
+        buf.writeItemStack(getLockedImport().getStackInSlot(0));
+        buf.writeBoolean(getLockedImport().isLocked());
+    }
+
+    @Override
+    public void receiveInitialSyncData(PacketBuffer buf) {
+        super.receiveInitialSyncData(buf);
+        try {
+            getLockedImport().setStackInSlot(0, buf.readItemStack());
+        } catch (IOException e) { // ignored
+        }
```

Comment:
buf.readItemStack() unfortunately throws one, and it doesn't make sense to either change the method signature or try to crash the game if it fails (the worst that could happen is that the following boolean read fails as a result, which would make it crash hard anyways; otherwise the import bus only acts slightly off), so I've just put a dummy try/catch here.